src/org/opensolaris/opengrok/analysis/Ctags.java

```diff
@@ -516,19 +535,43 @@ private void readTags(Definitions defs) {
                     //TODO if some languages use different character for separating arguments, below needs to be adjusted
                     String[] args = signature.split(",");
                     for (String arg : args) {
-                        //log.fine("Param = "+ arg);
-                        int space = arg.lastIndexOf(' ');//TODO this is not the best way, but works to find the last string(name) in the argument, hence skipping type
-                        if (space > 0 && space < arg.length()) {
-                            String afters = arg.substring(space + 1);
-                            //FIXME this will not work for typeless languages such as python or assignments inside signature ... but since ctags doesn't provide signatures for python yet and assigning stuff in signature is not the case for c or java, we don't care ...
-                            String[] names = afters.split("[\\W]"); //this should just parse out variables, we assume first non empty text is the argument name
-                            for (String name : names) {
-                                if (name.length() > 0) {
-                                    //log.fine("Param Def = "+ string);
-                                    addTag(defs, seenSymbols, lnum, name, "argument",
+                        //TODO this algorithm assumes that data types occur to
+                        //     the left of the argument name, so it will not
+                        //     work for languages like rust, kotlin, etc. which
```

Comment:
ok, this is not great,
if the old approach worked for kotlin, then I'd like to keep it
if it didn't and this is an improvement, then I am OK

Reply:
btw. this code is for parsing the ctags output and this format is given, so we don't need to assume anything or make it work as reverse engineering, format is here: http://docs.ctags.io/en/latest/news.html#reference-tags

Reply:
The old approach would not work for any language that has its data type to the right of the named argument. It was assuming 'C' style syntax.  <datatype> <argName>. It searched backward for that intervening space between datatype and argName. It also did not work for typeless languages. The algorithm I put into place will work for type less languages and datatypes to the left of the argName.  To get the parsing to work for languages like kotlin we need to get a ctag attribute in the language that tells us its data type description is to the right side of the argument.

PS. We do have to parse this because this is the signature string which is all the stuff between the function's parenthesis. Another issue can be raised to deal with this one.

Reply:
ok, so let's make this pass on full tests with universal ctags - @vladak - can you run this as a clone of a job on helius pointing to @shaehn s branch/fork ?
This way we will get a latest build with latest java and (if Vlada will update universal ctags) latest ctags and we will see if your changes break something
Unfortunately I don't have access to helius anymore and I didn't manage to build its copy outside of Oracle yet.

Reply:
I was finally able to get the build to go green.  I had to update the CtagsParserTest.java because the new parsing section I changed found more symbols. Some of the examples in the tests actually failed the old parsing method. 

Currently, this is not going to fail kotlin programs because there is no signature being gathered by the parser in Ctags.java